### DISCUSSION TITLE
createReport rate limit not refreshing?

### DISCUSSION THREAD

**Discussion by CraigMusser** [amzn/selling-partner-api-models#3686] [General] [answered]
The [createReport (POST /reports/2021-06-30/reports) Reports API documentation](https://developer-docs.amazon.com/sp-api/docs/reports-api-v2021-06-30-reference#post-reports2021-06-30reports) shows that the rate limit is 0.0167 requests per second with a burst of 15. That should be one request every 60 seconds. But I'm having an issue where I get rate limited for hours after using up the burst allotment. 

The HTTP response is 429 Too Many Requests with a response body of 
`{ "errors": [ { "code": "QuotaExceeded", "message": "You exceeded your quota for the requested resource.", "details": "" } ] }`

This doesn't seem to clear up until I try again the next day. Other API calls will work just fine.

**Comment by hhkwon** [comment]
> The [createReport (POST /reports/2021-06-30/reports) Reports API documentation](https://developer-docs.amazon.com/sp-api/docs/reports-api-v2021-06-30-reference#post-reports2021-06-30reports) shows that the rate limit is 0.0167 requests per second with a burst of 15. That should be one request every 60 seconds. But I'm having an issue where I get rate limited for hours after using up the burst allotment.
> 
> The HTTP response is 429 Too Many Requests with a response body of `{ "errors": [ { "code": "QuotaExceeded", "message": "You exceeded your quota for the requested resource.", "details": "" } ] }`
> 
> This doesn't seem to clear up until I try again the next day. Other API calls will work just fine.

Have you solved this problem? i got same issue.
Dear, Bro I just wonder when the limit will reset..

**Comment by felipeeflores** [comment]
Same here, in our case is when requesting the `GET_SALES_AND_TRAFFIC_REPORT` we get the header with the value set to `0.0167` (on successes only, terrible design by the way). We have a retry in place that attempts 4 times with exponential delays and we still get the same error (>20 mins later).

It seems as if we can only call it once an hour? Or perhaps every >30 minutes like the [FBA](https://developer-docs.amazon.com/sp-api/docs/report-type-values-fba) reports mentions? 

Anyone from the Amazon API to chip in here? This is really problematic, how can someone do anything useful if they need simple metrics like 30 and 365 day sales? it's really frustrating working on such a restrictive API.

**Comment by chrisskhoury** [comment]
Hello all! I am looking into this with the Reports team and will get back to you on this!

Best,
Chris Solution Architect SP API

**Reply by chrisskhoury** [reply, marked as answer]
The rate limit is now set to 3 requests per 5 minutes.

For future issues, the SP-API Health Dashboard is a great first stop: https://sellercentral.amazon.com/sp-api-status?locale=en-US
It gives you real-time status across all SP-API endpoints, which can save a lot of troubleshooting time.

Hope that gets you unblocked. Have a great start to the week! :)

Best,
Chris Khoury Solution Architect, SP API
